0507 rel notes (#3640)
Azure/AKS: CHANGELOG.md

# Azure Kubernetes Service Changelog
## Release 2023-05-07

Monitor the release status by regions at [AKS-Release-Tracker](https://releases.aks.azure.com/).

### Announcements
 

> * Starting on March 21, 2023, traffic to k8s.gcr.io will be redirected to registry.k8s.io, following the [community announcement](https://kubernetes.io/blog/2023/03/10/image-registry-redirect/).

Reviewer: We can remove this line. It's now more than a month

### Release notes

* Preview Features
   * Mariner is now supported in [NodeOSUpgradeChannel (preview)](https://learn.microsoft.com/azure/aks/auto-upgrade-node-image). This change is currently rolling out and expected to be in all regions by May 12th.

* Bug Fixes
   * Upgraded csi-provisioner version to v3.5.0 to fix a volume deletion issue, [details](https://github.com/kubernetes/kubernetes/issues/100485#issuecomment-1497878875)
   * Added update permission for ciliumidentity, without this permission- cilium-operator and ciliumAgent cannot garbage collect unused [identities](https://github.com/cilium/cilium/commit/2adf5f4313d42ae055528b09eb8dff4c19e74a98).
   * Now returning a clientError, when you do a Stop/RunCommand action on a cluster that was never successfully provisioned and was stuck in failed state. Error message returned is "The cluster is being deleted or hasn't been fully provisioned yet.".
   * Upgraded blob csi driver to [v1.19.5](https://github.com/kubernetes-sigs/blob-csi-driver/releases/tag/v1.19.5) on AKS 1.24, 1.25 to fix blobfuse install failures.

* Component Updates

>   * AKS updated the OSM addon on clusters running AKS k8s version >= v1.24.0 from OSM v1.2.3 to v.1.2.4. Details [here](https://github.com/openservicemesh/osm/releases/tag/v1.2.4).

Reviewer: ```suggestion
  * Open Service Mesh add-on images updated from v1.2.3 to [v1.2.4](https://github.com/openservicemesh/osm/releases/tag/v1.2.4) for AKS clusters of version >= 1.24.0.
```

>   * Azure Service Mesh add-on's Istiod and ingress images updated from v1.17.1 to [v1.17.2 patch](https://github.com/istio/istio/compare/1.17.1...1.17.2).

Reviewer: ```suggestion
  * Istio-based service mesh add-on's istiod and ingress images updated from v1.17.1 to v1.17.2. User needs to restart the workload pods to trigger re-injection of the newer patch version of istio-proxy. More information can be found [here](https://learn.microsoft.com/azure/aks/istio-upgrade).
```

>   * Upgraded Cilium to version 1.12.8 along with changes to match [upstream](https://github.com/cilium/cilium/pull/24075/files#diff-8d03af2c9f8f368b86831ae81cfb6f4152e884baf74366c09f2d92363ad791e7).

Reviewer: ```suggestion
  * Cilium upgraded to [1.12.8](https://github.com/cilium/cilium/releases/tag/v1.12.8) for [AKS clusters with Azure CNI Powered by Cilium](https://learn.microsoft.com/en-us/azure/aks/azure-cni-powered-by-cilium).
```

   * The CPU limit of Windows DaemonSet for Azure Monitor Metrics Addon is updated from 200m to 500m to fix throttling issue.

* Component Updates
  * Open Service Mesh add-on images updated from v1.2.3 to [v1.2.4](https://github.com/openservicemesh/osm/releases/tag/v1.2.4) for AKS clusters of versions >= 1.24.0.
  * Istio-based service mesh add-on's istiod and ingress images updated from v1.17.1 to v1.17.2. User needs to restart the workload pods to trigger re-injection of the newer patch version of istio-proxy. More information can be found [here](https://learn.microsoft.com/azure/aks/istio-upgrade).
  * Cilium upgraded to [1.12.8](https://github.com/cilium/cilium/releases/tag/v1.12.8) for [AKS clusters with Azure CNI Powered by Cilium](https://learn.microsoft.com/en-us/azure/aks/azure-cni-powered-by-cilium).
  * Blob csi driver upgraded to [v1.19.5](https://github.com/kubernetes-sigs/blob-csi-driver/releases/tag/v1.19.5) on AKS 1.24, 1.25 to fix blobfuse install failures.
  * Csi-provisioner version updated to v3.5.0 in order to fix a volume deletion issue, [details](https://github.com/kubernetes/kubernetes/issues/100485#issuecomment-1497878875)

>   * Validations logic added to error out in [Azure CNI overlay](https://learn.microsoft.com/en-us/troubleshoot/azure/azure-kubernetes/insufficientsubnetsize-error-advanced-networking) clusters, whenever autoscaler max count or manual scale node count likely goes over the IP limits of what the managed subnet and podcidr settings get by default

Reviewer 2: Why is the validation a component update and not a bug fix or behavioral change?

Reviewer 2: > whenever autoscaler max count or manual scale node count likely goes over the IP limits of what the managed subnet and podcidr settings get by default

also what does this mean? What happened before that now doesn't happen or vice versa?

> 

Reviewer 2: Where's the VHDs?

### Release notes

* Preview Features
   * Mariner is now supported in [NodeOSUpgradeChannel (preview)](https://learn.microsoft.com/azure/aks/auto-upgrade-node-image). This change is currently rolling out and expected to be in all regions by May 12th.

* Bug Fixes
   * Upgraded csi-provisioner version to v3.5.0 to fix a volume deletion issue, [details](https://github.com/kubernetes/kubernetes/issues/100485#issuecomment-1497878875)
   * Added update permission for ciliumidentity, without this permission- cilium-operator and ciliumAgent cannot garbage collect unused [identities](https://github.com/cilium/cilium/commit/2adf5f4313d42ae055528b09eb8dff4c19e74a98).

>    * Now returning a clientError, when you do Stop/RunCommand action on a cluster that was never successfully provisioned and was stuck in failed state.

Reviewer: do we have the exact error message?

>    * Upgraded blob csi driver to [v1.19.5](https://github.com/kubernetes-sigs/blob-csi-driver/releases/tag/v1.19.5) on AKS 1.24, 1.25 to fix blobfuse install failures.

Reviewer: this should be under component updates section

* Behavior Changes
   * The maximum size of a request body for Nginx ingress controller default increased from 1MB to 64MB.Refer ARM specs [here](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#max-request-body-size).

>    * The Docker Bridge CIDR field in the AKS API was made redundant during our change from Docker to containerD in Kubernetes version 1.19. Starting with the 2023-04-01 AKS API version, the Docker Bridge CIDR field will be removed. 

Reviewer: should this be under announcements?

Author: Good point.

### Announcements
 

> * Docker container runtime will be retired for Windows nodepools on May 1, 2023. After docker container runtime is retired, you may remain on existing deployed instances but scaling operations will fail, nodepool creation will fail, and you will be out of support. Follow the detailed steps [in our documentation](https://learn.microsoft.com/azure/aks/learn/quick-windows-container-deploy-cli) to upgrade to containerd.

Reviewer: nit for tense: "Docker container runtime will be retired for Windows nodepools on May 1, 2023" -> "Docker container runtime has been retired for Windows nodepools on May 1, 2023"

   * Mariner is now supported in [NodeOSUpgradeChannel (preview)](https://learn.microsoft.com/azure/aks/auto-upgrade-node-image). This change is currently rolling out and expected to be in all regions by May 12th.

* Bug Fixes

>    * Csi-provisioner version updated to v3.5.0 in order to fix a volume deletion issue, [details](https://github.com/kubernetes/kubernetes/issues/100485#issuecomment-1497878875)

Reviewer: this is component version bump. we need to move it there

>    * Added update permission for ciliumidentity, without this permission- cilium-operator and ciliumAgent cannot garbage collect unused [identities](https://github.com/cilium/cilium/commit/2adf5f4313d42ae055528b09eb8dff4c19e74a98).

Reviewer: Updated 'cilium', 'cilium-operator', 'cilium-pre-flight' ClusterRoles to include 'update' permission for 'ciliumidentities' api-resource. This addresses the issue where cilium-operator and cilium-agent couldn't garbage collect unused identities.

   * Now returning a clientError, when you do a Stop/RunCommand action on a cluster that was never successfully provisioned and was stuck in failed state. Error message returned is "The cluster is being deleted or hasn't been fully provisioned yet.".

>    * The CPU limit of Azure Monitor Metrics Addon (Windows Daemonset) is updated from 200m to 500m to fix a throttling issue.

Reviewer: The CPU limit Windows DaemonSet ofAzure Monitor Metrics Addon is updated from 200m to 500m to fix throttling issue.
